Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp

```diff
@@ -585,6 +569,74 @@ namespace Multiplayer
         stats.RecordFrameTime(AZ::TimeUs{ duration.count() });
     }
 
+    void MultiplayerSystemComponent::UpdatedMetricsConnectionCount()
+    {
+        MultiplayerStats& stats = GetStats();
+        auto sendNetworkUpdates = [&stats](IConnection& connection)
+        {
+            if (connection.GetUserData() != nullptr)
+            {
+                IConnectionData* connectionData = reinterpret_cast<IConnectionData*>(connection.GetUserData());
+                if (connectionData->GetConnectionDataType() == ConnectionDataType::ServerToClient)
+                {
+                    stats.m_clientConnectionCount++;
+                }
+                else
+                {
+                    stats.m_serverConnectionCount++;
+                }
+            }
+        };
+        m_networkInterface->GetConnectionSet().VisitConnections(sendNetworkUpdates);
+    }
+
+    void MultiplayerSystemComponent::UpdateConnections()
+    {
+        if (GetAgentType() == MultiplayerAgentType::ClientServer || GetAgentType() == MultiplayerAgentType::DedicatedServer)
+        {
+            // Threaded update calls.
+            if (m_networkInterface->GetConnectionSet().GetConnectionCount() > 0)
+            {
+                AZ_PROFILE_SCOPE(MULTIPLAYER, "MultiplayerSystemComponent: UpdateConnections");
+
+                AZ::JobCompletion jobCompletion;
+
+                auto sendNetworkUpdates = [&jobCompletion](IConnection& connection)
+                {
+                    AZ::Job* job = AZ::CreateJobFunction([&connection]()
+                        {
+                            if (connection.GetUserData() != nullptr)
+                            {
+                                IConnectionData* connectionData = static_cast<IConnectionData*>(connection.GetUserData());
+                                connectionData->Update();
+                            }
+                        }, true /*auto delete*/, nullptr);
+
+                    job->SetDependent(&jobCompletion);
+                    job->Start();
+                };
+
+                m_networkInterface->GetConnectionSet().VisitConnections(sendNetworkUpdates);
+                jobCompletion.StartAndWaitForCompletion();
+            }
+        }
+        else // On clients (including the Editor) run in a single threaded mode to avoid issues in UI asset loading
```

Comment:
can you provide more details on the UI asset loading issues?

Reply:
I tried getting a repro for this but now client connections dont even connect to the server if they were updated from a thread. So there is something else blocking this. But this isn't necessary for performance work.